src/libs/record/record.c

```diff
@@ -0,0 +1,780 @@
+#include <kdbchangetracking.h>
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <kdbprivate.h>
+#include <kdbrecord.h>
+
+/**
+ * Enable session recording.
+ * This affects both the given @p handle as well as every KDB instance that will be created after this method has been called.
+ *
+ * @param handle the KDB instance to use
+ * @param parentKey recording will be enabled for every key that is same or below this key
+ * @param errorKey used for reporting errors and warnings.
+ *                 As usual, they will be found as meta keys attached to this key.
+ * @retval true - recording has been enabled successfully
+ * @retfal false - there was an error enabling recording - see @p errorKey for further details
+ */
+bool elektraRecordEnableRecording (KDB * handle, const Key * parentKey, Key * errorKey)
+{
+	if (handle == NULL)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "NULL pointer passed for KDB handle");
+		return false;
+	}
+
+	if (parentKey == NULL)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "NULL pointer passed for parent key");
+		return false;
+	}
+
+	if (handle->hooks.record.plugin == NULL)
+	{
+		ELEKTRA_ADD_RESOURCE_WARNING (
+			errorKey, "There is no record plugin present. Session recording will not work for the current KDB instance.");
+	}
+
+	Key * configKey = keyNew (ELEKTRA_RECORD_CONFIG_KEY, KEY_END);
+
+	KeySet * config = ksNew (0, KS_END);
+
+	if (kdbGet (handle, config, configKey) == -1)
+	{
+		elektraCopyError (errorKey, configKey);
```

Comment:
Yeah that's true. Because copying warnings would require array handling 😅.

Reply:
Ah yes, if you want to keep existing warnings that's more complicated. You don't really need full array handling since the warning array is limited to 100 entries, but you still need something...

That's also (partly) why `kdbGet`/`kdbSet` has `initialParent`. You could do something similar. 

Replacing existing warnings would just be `ksAppend (..., ksBelow())`, so that might a partial solution. 

Loosing all warnings from `kdbGet`/`kdbSet` could be a problem since most errors from `kdbGet`/`kdbSet` are pretty hard to diagnose without the warnings. IMO it might be better to replace existing warnings so at least error can be understood.

Reply:
Well as the warnings array keys are pretty much fixed (no fancy edgecases), I think a simple merge logic isn't that hard to implement.

Reply:
Yup, I already have an almost working version of a `elektraAddWarningsFrom` function...

Reply:
I just utilize the already existing `addWarning` function now 😅